src/Rule/Date.php

```diff
@@ -186,54 +189,83 @@ class Date extends Rule
         'full' => 0, // IntlDateFormatter::FULL,
     ];
 
-    public function __construct()
+    /**
+     * Date constructor.
```

Comment:
It was meant to split this class into individual ones. There's `TODO: split into 3 separate classes`